iree/turbine/kernel/compiler/host_codegen.py

```diff
@@ -55,7 +55,8 @@ def isolated_test_call(
         argument_dims = get_dynamic_dims(
             sig.kernel_buffer_input_bindings, dynamic_symbols
         )
-        input_tensors += [IndexType.get() for _ in argument_dims]
+        # Only add types for the unique dynamic symbols.
+        input_tensors += [IndexType.get() for _ in set(argument_dims)]
```

Comment:
funny, I also have this in my zero copy PR to handle dynamic writable argument 😁

Reply:
yes its a bug and I will change to preserve insertion order.